src/server/services/afis/afis.ts

```diff
@@ -234,3 +244,223 @@ export async function fetchAfisBusinessPartnerDetails(
+    return 'open';
+  } else if (dunningLevel === 0 && sepaMandate) {
+    return 'automatische-incasso';
+  } else if (dunningBlockingReason === 'D') {
+    return 'dispuut';
+  }
+
+  const invoiceStatus = null;
+  // RP TODO: Tellen dit als persoonsgegevens? Zijn de praktische overwegingen genoeg om dit te loggen?
+  captureMessage(
+    `invoiceStatus could not be determined; setting invoiceStatus to '${invoiceStatus}'\n` +
+      `\tDunningBlockingReason = ${dunningBlockingReason}, SEPAMandata = ${sepaMandate}`,
+    { severity: 'error' }
+  );
+  return invoiceStatus;
+}
+
+export async function fetchAfisClosedInvoices(
+  requestID: RequestID,
+  businessPartnerID: number,
+  top?: number
+) {
+  const fetchClosed = async (
+    requestID: RequestID,
+    config: DataRequestConfig
+  ) => {
+    const filter = `&$filter=Customer eq '${businessPartnerID}' and IsCleared eq true and (DunningLevel eq '0' or ReverseDocument ne '')`;
+    const select =
+      '$select=ReverseDocument,ProfitCenterName,DunningLevel,InvoiceNo,NetDueDate';
+    const orderBy = '$orderBy=NetDueDate desc';
+
+    let closedInvoicesQuery = `${filter}&${select}&${orderBy}`;
+    config.url = `${config.url}${closedInvoicesQuery}`;
+
+    config.transformResponse = (data: AfisCloseInvoiceSource) =>
+      getFeedEntryProperties(data).map((invoiceProperties) =>
+        transformClosedInvoices(businessPartnerID, invoiceProperties)
+      );
+
+    const response = await requestData<AfisClosedInvoice[]>(config, requestID);
+    return response;
+  };
+
+  return await fetchAfisFacturen(
+    fetchClosed,
+    requestID,
+    businessPartnerID,
+    top
+  );
+}
+
+function transformClosedInvoices(
+  businessPartnerID: number,
+  fields: AfisClosedInvoicePropertiesSource
+): AfisClosedInvoice {
+  const [invoiceNoEncrypted] = encrypt(
+    `${businessPartnerID}:${fields.InvoiceNo}`
+  );
+  return {
+    profitCenterName: fields.ProfitCenterName,
+    dueDate: fields.NetDueDate,
+    dueDateFormatted: defaultDateFormat(fields.NetDueDate),
+    invoiceNoEncrypted,
+    invoiceStatus: determineClosedInvoiceStatus(
+      fields.DunningLevel,
+      fields.ReverseDocument
+    ),
+  };
+}
+
+function determineClosedInvoiceStatus(
+  dunningLevel: AfisClosedInvoicePropertiesSource['DunningLevel'],
+  reverseDocument: AfisClosedInvoicePropertiesSource['ReverseDocument']
+): AfisClosedInvoice['invoiceStatus'] {
+  if (reverseDocument) {
+    return 'geannuleerd';
+  } else if (dunningLevel === 0) {
+    return 'betaald';
+  }
+
+  const invoiceStatus = null;
+  // RP TODO: Tellen dit als persoonsgegevens? Zijn de praktische overwegingen genoeg om dit te loggen?
+  captureMessage(
+    `invoiceStatus could not be determined; setting invoiceStatus to '${invoiceStatus}'\n` +
+      `\treverseDocument was ${reverseDocument ? 'found' : 'not found'}, DunningLevel = ${dunningLevel}`,
+    { severity: 'error' }
+  );
+  return invoiceStatus;
+}
+
+async function fetchAfisFacturen(
```

Comment:
Ik denk dat we hier nog veel kunnen versimpelen door geen onderscheid te maken tussen de fetchopen / fetchclosed services (net als 1 type voor alle invoices). Je kunt dan functionaliteit samenvoegen, versimpelen en leesbaarder maken.

Een voorbeeld van een samengevoegd geheel zou kunnen zijn:

```
type AfisInvoiceStatus =
  | 'closed'
  | 'open'
  | 'cancelled'
  | 'disputed'
  | 'automatische-incasso' // NL naam omdat deze term sterk gekoppeld is aan onze taal. 'e-mandate' zou ook kunnen maar ik neig naar de nl term.
  | 'overgedragen-aan-belastingen'; // Ook deze term nederlands vanwege specificiteit

interface AfisInvoice {
  id: string;
  sender: string; // profitCenterName
  datePublished: string; // postingDate?
  datePublishedFormatted: string;
  dueDate: string;
  dueDateFormatted: string;
  amount: string | null;
  amountFormatted: string | null;
  status: AfisInvoiceStatus;
  payLink: string | null;
  documentDownloadLink: string | null; // bij overgedragen-aan-belastingen geen downloads?
}

type AfisInvoiceSource { }

function transformAfisInvoicesResponse(sessionID: AuthProfileAndToken['profile']['sid'], afisInvoiceResponseData: AfisApiFeedResponseSource<AfisInvoiceSource>) {
  const feedEntries = getFeedEntryProperties(afisInvoiceResponseData);
  if (feedEntries.length) {
    return feedEntries.map((invoiceSource) => {
      const afisInvoiceTransformed: AfisInvoice = {
        id: '',
        sender: '',
        datePublished: '',
        datePublishedFormatted: '',
        dueDate: '',
        dueDateFormatted: '',
        amount: '',
        amountFormatted: '',
        status: deterimiseAfisInvoiceStatus(invoiceSource),
        payLink: '',
        documentDownloadLink: generateAfisDownloadLink(sessionID, invoiceSource), // Generates full url to download endpiint
      }
      return afisInvoiceTransformed;
    });
  }
}

type AfisInvoiceQueryOptions = {
  skip?: string;
  top?: string;
  status: AfisInvoiceStatus;
}

async function fetchAfisInvoices(requestID: RequestID, authProfileAndToken: AuthProfileAndToken, options: AfisInvoiceQueryOptions) {

  const afisOdataFilterQuerySpecific = getOdataInvoiceFilter(authProfileAndToken.profile.id, options.status)

  const requestConfig = getApiConfig('AFIS', {
    formatUrl: (config) => {
      return `${config.url}/${afisOdataSelectQueryCommon}${afisOdataFilterQuerySpecific}${afisOdataOrderQueryCommon}`
    }
  });

  return requestData<AfisInvoice>({ ...requestConfig, transformResponse: transformAfisInvoicesResponse }, requestID);
}
```

Reply:
Het gaat hier niet om bovenstaande code te gebruiken maar om een en ander wat compacter te maken. Hou 't simpel.

Reply:
En `documentDownloadLink: generateAfisDownloadLink(sessionID, invoiceSource), // Generates full url to download endpiint`

Hierin moet je dus iets doen als.
```
const idEncrypted = encrypt(`${sessionID}:${invoiceNo}`);
generateFullApiUrlBFF(BffEndpoints.AFIS_DOCUMENT_DOWNLOAD,{ id: idEncrypted });
```

Dus mét session id voor added security